web/src/components/FeedbackContainer.tsx

```diff
@@ -0,0 +1,162 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import TextButton from './TextButton'
+
+const IconTextContainer = styled.div`
+  margin-top: 30px;
+  width: 340px;
+  padding: 0 30px;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`
+const Text = styled.div`
+  padding: 10px;
+  text-align: center;
+`
+
+type PropsType = {
+  cityCode: string
+  language: string
+  routeType: RouteType
+  isPositiveFeedback: boolean
+  isSearchFeedback: boolean
+  closeModal?: () => void
+  alias?: string
+  path?: string
+  query?: string
+}
+
+export type SendingStatusType = 'IDLE' | 'SUCCESS' | 'ERROR' | 'SENDING'
+
+export const FeedbackContainer = (props: PropsType): ReactElement => {
+  const [comment, setComment] = useState<string>('')
+  const [contactMail, setContactMail] = useState<string>('')
+  const [sendingStatus, setSendingStatus] = useState<SendingStatusType>('IDLE')
+  const { path, alias, query, language, isPositiveFeedback, isSearchFeedback, routeType, cityCode, closeModal } = props
+  const { t } = useTranslation('feedback')
+
+  const getFeedbackType = (): FeedbackType => {
+    switch (routeType) {
+      case EVENTS_ROUTE:
+        return path ? PAGE_FEEDBACK_TYPE : EVENTS_FEEDBACK_TYPE
+
+      case OFFERS_ROUTE:
+        return alias ? OFFER_FEEDBACK_TYPE : OFFERS_FEEDBACK_TYPE
+
+      case DISCLAIMER_ROUTE:
+        return PAGE_FEEDBACK_TYPE
+
+      case POIS_ROUTE:
+        return path ? PAGE_FEEDBACK_TYPE : CATEGORIES_FEEDBACK_TYPE
+
+      case CATEGORIES_ROUTE:
+        return path ? PAGE_FEEDBACK_TYPE : CATEGORIES_FEEDBACK_TYPE
+
+      case SEARCH_ROUTE:
+        return SEARCH_FEEDBACK_TYPE
+
+      default:
+        return CATEGORIES_FEEDBACK_TYPE
+    }
+  }
+
+  const getFeedbackData = (comment: string, contactMail: string): FeedbackParamsType => {
+    const feedbackType = getFeedbackType()
+    const commentWithMail = `${comment}    Kontaktadresse: ${contactMail || 'Keine Angabe'}`
+    return {
+      feedbackType,
+      feedbackCategory: CONTENT_FEEDBACK_CATEGORY,
+      isPositiveRating: isPositiveFeedback,
+      permalink: path,
+      city: cityCode,
+      language,
+      comment: commentWithMail,
+      alias,
+      query
+    }
+  }
+
+  const onFeedbackCommentChanged = (comment: string) => setComment(comment)
+
+  const onFeedbackContactMailChanged = (contactMail: string) => setContactMail(contactMail)
+
+  const handleSubmit = () => {
+    const feedbackData = getFeedbackData(comment, contactMail)
+    setSendingStatus('SENDING')
+
+    const request = async () => {
+      const apiUrl = cmsApiBaseUrl
+      const feedbackEndpoint = createFeedbackEndpoint(apiUrl)
+      await feedbackEndpoint.request(feedbackData)
+      setSendingStatus('SUCCESS')
+    }
+
+    request().catch(err => {
+      console.log(err)
+      setSendingStatus('ERROR')
+    })
+  }
+
+  if (['IDLE', 'ERROR', 'SENDING'].includes(sendingStatus)) {
```

Comment:
Switching the if would make it a little bit easier imo, i.e. checking for `sendingStatus === 'SUCCESS`.